src/ty/mod.rs

```diff
@@ -576,17 +576,21 @@ where
     }
 }
 
-#[cfg(feature = "bit-vec")]
 impl TypeDefBitSequence {
     /// Creates a new [`TypeDefBitSequence`] for the supplied bit order and bit store types.
-    pub fn new<T, O>() -> Self
+    ///
+    /// With the `bit-vec` feature enabled, the expected usage is to provide either
+    /// `bitvec::order::Lsb0` or `bitvec::order::Msb0` as the order type, and then something
+    /// like u8, u8, or u32 as the store type. Without the `bit-vec` feature enabled, it's
+    /// recommended that your types have identical `TypeInfo` to those.
```

Comment:
…or else? :)
Like, is "recommended" the right word here? Isn't it more "has to be"?

Reply:
I mean, in theory it can be whatever it wants; it's just that our current tooling won't know how to decode the thing if the types don't line up with what we expect (no reason somebody else can't make tooling that can support more exotic types though if they want to go that route)